handlers.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/rolandovlz/goauth/models"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func loginHandler(w http.ResponseWriter, r *http.Request) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	user := &models.User{}
+
+	err = json.Unmarshal(body, user)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	row := db.QueryRow("SELECT password FROM users WHERE username=$1;", user.Username)
+
+	var hashFromDb string
+
+	switch err = row.Scan(&hashFromDb); err {
+	case sql.ErrNoRows:
+		log.Println("Not found!")
+		http.Redirect(w, r, "/login", http.StatusBadRequest)
+	case nil:
+		if err := bcrypt.CompareHashAndPassword([]byte(hashFromDb), []byte(user.Password)); err != nil {
```

Comment:
Is there a need for assigning  defining err in this line? Can't the same by accomplished with:  

``bcrypt.CompareHashAndPassword([]byte(hashFromDb), []byte(user.Password)) != nil``?